Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,1111 @@
+        0x58D2C5,            // 63 MissionTitle text
+        NULL,                // 0x58D447, // 64 MissionTitle text
+        0x58CBAB,            // 65 WastedBusted text
+        NULL,                // 0x58B1B7, // 66 Timers
+        0x58B263,            // 67 Timers
+        NULL,                // 0x58B435, // 68 Timers
+        NULL,                // 0x58B536, // 69 Timers
+        0x58B5DE,            // 70 Timers
+        NULL,                // 71
+        NULL,                // 72
+        0x58B7BD,            // 73 Help text
+        0x58BA4C,            // 74 Help text
+        0x58BBA7,            // 75 Help text
+        0x58BD19,            // 76 Help text
+        0x58BE2B,            // 77 Help text
+        0x58BF1C,            // 78 Help text
+        0x58BFCB,            // 79 Help text
+        NULL,                // 80
+        NULL,                // 81
+        NULL,                // 82
+        0x580E11,            // 83 Menu system
+        0x580F85,            // 84
+        0x5810CC,            // 85
+        0x581132,            // 86
+        0x5811A1,            // 87
+        0x58147A,            // 88
+        0x5814E7,            // 89
+        0x581699,            // 90
+        NULL,                // 91
+        NULL,                // 92
+        NULL,                // 93
+        0x581889,            // 94
+        0x58CCC5,            // 95 OddJob
+        0x58CDD0,            // 96 OddJob
+        0x58CECC,            // 97 OddJob
+        0x58D146,            // 98 OddJob
+        NULL,                // 0x58A199, // 99 TripSkip
+        NULL,                // 0x58A207, // 100 TripSkip
+        NULL,                // 0x58A2B0, // 101 TripSkip
+        0x4E9F1A,            // 102 RadioStation
+        0x43CF47,            // 103 CDarkel
+        0x4477B7,            // 104 CGarages
+        0x4478AC,            // 105 CGarages
+    };
+
+    for (int i = 0; i < sizeof(m_dwHUDWidth) / sizeof(const void*); i++)
+    {
+        if (m_dwHUDWidth[i] != NULL)
+            MemPut<const void*>(m_dwHUDWidth[i] + 0x2, &fHUDWidth[i]);
+    }
+
+    for (int i = 0; i < sizeof(m_dwHUDHeight) / sizeof(const void*); i++)
+    {
+        if (m_dwHUDHeight[i] != NULL)
+            MemPut<const void*>(m_dwHUDHeight[i] + 0x2, &fHUDHeight[i]);
+    }
+
+    /*
+    // Help text bar chart offset
+    static float fBarChartOffsetY = 160.0f;
+    MemPut<const void*>(0x58BE9F + 0x2, &fBarChartOffsetY);
+
+    // Lock Subtitles Width
+    static float fSubtitlesMult = 1.0f;
+    MemPut<const void*>(0x58C4E8 + 0x2, &fSubtitlesMult);
+    */
+
+    // Second player fix.
+    MemPut<const void*>(0x58F9A0 + 0x2, &fHUDWidth[110]);            // Weapon icon X
+    MemPut<const void*>(0x58F993 + 0x2, &fHUDWidth[16]);             // Weapon icon X
+    MemPut<const void*>(0x58F972 + 0x2, &fHUDHeight[16]);            // Weapon icon Y
+    MemPut<const void*>(0x58FA8E + 0x2, &fHUDWidth[17]);             // Ammo x
```

Comment:
Use type float. Why do you put the addresses of the values ​​in memory instead of the values ​​themselves directly?